pkg/tcpip/transport/icmp/protocol.go

```diff
@@ -124,6 +124,14 @@ func (*protocol) Close() {}
 // Wait implements stack.TransportProtocol.Wait.
 func (*protocol) Wait() {}
 
+// TODO(gvisor.dev/issue/170): ICMP packets don't currently get parsed early
+// because it's common for netstack users not to enable ICMP endpoints, but to
+// still expect netstack to handle ICMP correctly. Support for parsing
+// protocols and creating endpoints for that protocol should be decoupled.
+func (*protocol) Parse(pkt *tcpip.PacketBuffer, netProto tcpip.NetworkProtocolNumber) {
```

Comment:
I am confused why this is empty. I read the comment but it still didn't make sense to me:(